modules/pricer/src/main/java/com/opengamma/strata/pricer/credit/cds/IsdaCdsProductPricer.java

```diff
@@ -0,0 +1,743 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.pricer.credit.cds;
+
+import static com.opengamma.strata.math.impl.util.Epsilon.epsilon;
+import static com.opengamma.strata.math.impl.util.Epsilon.epsilonP;
+import static com.opengamma.strata.math.impl.util.Epsilon.epsilonPP;
+
+import java.time.LocalDate;
+
+import com.opengamma.strata.basics.ReferenceData;
+import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.basics.currency.CurrencyAmount;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.collect.array.DoubleArray;
+import com.opengamma.strata.collect.tuple.Pair;
+import com.opengamma.strata.market.sensitivity.PointSensitivityBuilder;
+import com.opengamma.strata.math.impl.util.Epsilon;
+import com.opengamma.strata.product.credit.CreditCouponPaymentPeriod;
+import com.opengamma.strata.product.credit.ResolvedCds;
+
+/**
+ * Pricer for single-name credit default swaps (CDS) based on ISDA standard model. 
+ * <p>
+ * The implementation is based on the ISDA model versions 1.8.2.
+ */
+public class IsdaCdsProductPricer {
+
+  /**
+   * Default implementation.
+   */
+  public static final IsdaCdsProductPricer DEFAULT = new IsdaCdsProductPricer(AccrualOnDefaultFormulae.ORIGINAL_ISDA);
+
+  /**
+   * The formula
+   */
+  private final AccrualOnDefaultFormulae formula;
+
+  /**
+   * The omega parameter.
+   */
+  private final double omega;
+
+  /**
+   * Constructor specifying the formula to use for the accrued on default calculation.  
+   * <p>
+   * Options are the formula given in the ISDA model (version 1.8.2 and lower); 
+   * the proposed fix by Markit (given as a comment in  version 1.8.2), or the mathematically correct formula. 
+   * <p>
+   * @param formula  the formula
+   */
+  public IsdaCdsProductPricer(AccrualOnDefaultFormulae formula) {
+    this.formula = ArgChecker.notNull(formula, "formula");
+    if (formula == AccrualOnDefaultFormulae.ORIGINAL_ISDA) {
+      omega = 1d / 730d;
+    } else {
+      omega = 0d;
+    }
```

Comment:
Could add a method `getOmega()` to `AccrualOnDefaultFormulae` to avoid the if statement.